### PR TITLE
BUILD: use numpy-wheels/openblas_support.py to create _distributor_init.py

### LINKED ISSUE #13764: BUILD: clean up handling of dll loading on Windows
In ~#13109~ #13019 we forced preloading of support DLLs on windows. As per the discussion that starts with [this comment](https://github.com/numpy/numpy/pull/13019#issuecomment-467694625). We should move the code to a `numpy/_distributor_init.py` file in [MacPython/numpy-wheels](https://github.com/MacPython/numpy-wheels)  and wrap it in a try-except in case `__file__` does not exist.

This issue should be resolved before 1.17

xref gh-12994

### PR DESCRIPTION
**Pull request by mattip**
WIP until ~MacPython/numpy-wheels#50 is merged~ MacPython/numpy-wheels#49 is fixed since it downloads the file from there to copy wheel-building behaviour (on windows CI runs only).

Fixes #13764, reverts parts of #13019

### CODE REVIEW

**Comment by charris** [comment]
The numpy-wheels PR has been merged.

**Comment by mattip** [comment]
Turns out MacPython/numpy-wheels#50 did not actually vendor in a `_distributor_init.py` file, so MacPython/numpy-wheels#51 was needed.

**Comment by mattip** [comment]
... and also MacPython/numpy-wheels#52. Sorry.

**Comment by charris** [comment]
What is with the failing tests?

**Comment by mattip** [comment]
The tests fail because this PR moves dll preloading from `numpy/core/__init__.py` to `_distributor_init.py`, which is downloaded from  MacPython/numpy-wheels, and it needs the fix in MacPython/numpy-wheels#52 to work properly.

**Comment by mattip** [comment]
Even the LGTM tests now pass, I am removing the WIP tag.

**Comment by mattip** [comment]
@charris ping

**Comment by charris** [comment]
Why get the file from `numpy-wheels`? It seems odd to have a test dependency there, why not put the in the `tools` directory?

**Comment by tylerjereddy** [comment]
> Why get the file from numpy-wheels? It seems odd to have a test dependency there, why not put it in the tools directory?

Yeah, I found it odd too, but I think Ralf and @cgohlke argued that it is a matter for distributors to decide on & so shouldn't be in the NumPy source proper. I think I made the same argument to push back against the migration of the infrastructure outside our source, since I couldn't really see the "harm" it was doing.

**Comment by charris** [comment]
@tylerjereddy Hmm. I don't see how putting it in `tools` would make it a necessary part of the build. Maybe I'm missing something...

**Comment by tylerjereddy** [comment]
@charris I suppose that could work if worth the effort to change from what is proposed. Note that the "test" that needs this file is isolated to the CI infrastructure itself on Azure & not in unit tests.

Hopefully we're sure whatever we do here is harmless--the original scenario was the source of quite a few issues popping up.

**Comment by rgommers** [comment]
I think either way is fine. In the `tools` dir would work. It doesn't really matter too much imho, in one case we duplicate code, in the other we grab something from another repo. 

> Yeah, I found it odd too, but I think Ralf and @cgohlke argued that it is a matter for distributors to decide on & so shouldn't be in the NumPy source proper

it definitely should not be where it is right now, in `tools` would be a different story though.

File: tools/openblas_support.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import os
+import textwrap
+
+def make_init(dirname):
+    '''
+    Create a _distributor_init.py file for OpenBlas
+    '''
+    with open(os.path.join(dirname, '_distributor_init.py'), 'wt') as fid:
```
Comment (charris) [inline tools/openblas_support.py]: Why not make this the file in tools and copy it into the root?

Comment (charris) [reply tools/openblas_support.py]: That is, name this `_distributor_init.py`.

Comment (charris) [reply tools/openblas_support.py]: The command might be
```
Copy-Item -Path tools\_distributor_init.py -Destination .
```

Comment (mattip) [reply tools/openblas_support.py]: My goal is to expand this file to include the download-and-expand parts of getting OpenBlas from rackspace, so we don't have separate stanzas in each CI system. Right now each OS on each CI system handles this differently, there is no single place that specifies the OpenBLAS version and root fo the rackspace directory. So it seems natural to keep all the OpenBLAS handling in one file

**Comment by charris** [comment]
Thanks Matti.

**Pull request merged**